### PR TITLE
Refactor rulesets send nosp

### PR DESCRIPTION
**Pull request by elena-vi**
# Context

Slowly slowly refactoring out the V2 engine into separate files, with goal of making the classification extensible.

## Changes proposed in this pull request

Move send NOSP classification into it's own file and test related helpers

## Link to Jira card

https://hackney.atlassian.net/browse/MAAP-201

## Things to check

- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] Environment variables have been updated

### CODE REVIEW

File: spec/lib/hackney/income/tenancy_classification/classifier_spec.rb [hunk 1]
```diff
@@ -145,14 +145,6 @@
...
     let(:unused_action_codes_required_for_uh_criteria_sql) { result - action_codes }
 
-    describe '#valid_actions_for_nosp_to_progress' do
-      let(:result) { assign_classification.send(:valid_actions_for_nosp_to_progress) }
-
-      it 'contains Letter 2 UH code that is used for an edge case in the UH Criteria SQL' do
-        expect(unused_action_codes_required_for_uh_criteria_sql).to eq([Hackney::Tenancy::ActionCodes::INCOME_COLLECTION_LETTER_2_UH])
-      end
-    end
-
```
Comment (Miles-Alford) [inline spec/lib/hackney/income/tenancy_classification/classifier_spec.rb:155]: Why remove these tests?, saw the same thing on the last PR. Think some other refactored classifications still have tests for their action codes, will remove them if we think its best to just remove them

Comment (elena-vi) [reply spec/lib/hackney/income/tenancy_classification/classifier_spec.rb:155]: these now live in a private method within that classification rule, so I wasn't sure how appropriately test these now

Comment (Miles-Alford) [reply spec/lib/hackney/income/tenancy_classification/classifier_spec.rb:155]: 👍

Comment (elena-vi) [reply spec/lib/hackney/income/tenancy_classification/classifier_spec.rb:155]: I made a card under the epic to address this later https://hackney.atlassian.net/browse/MAAP-226

**Pull request merged**